irohad/validation/stateful_validator_common.hpp

```diff
@@ -6,32 +6,38 @@
 #ifndef IROHA_STATEFUL_VALIDATOR_COMMON_HPP
 #define IROHA_STATEFUL_VALIDATOR_COMMON_HPP
 
-#include "common/types.hpp"
+#include <vector>
+#include "interfaces/common_objects/types.hpp"
 
 namespace shared_model {
   namespace interface {
     class Proposal;
   }
-}
+}  // namespace shared_model
 
 namespace iroha {
   namespace validation {
 
-    /// Name of the failed command
-    using CommandName = std::string;
+    /// Type of command error report
+    struct CommandError {
+      /// Name of the failed command
+      std::string name;
 
-    /// Error, with which the command failed
-    using CommandError = std::string;
+      /// Error, with which the command failed
+      std::string error;
 
-    /// Failed command's name and error
-    using CommandNameAndError = std::pair<CommandName, CommandError>;
+      /// Shows, if transaction has passed initial validation
+      bool tx_passed_initial_validation;
```

Comment:
What do you mean by initial?

Reply:
There are two kinds of tx validation: initial, when we check the tx itself (for example, existence of account_id of a creator) in StatefulValidator::validate, and the second, when we check it command-by-command